### ISSUE TITLE
[V2] Windows - runtime.WindowMaximise() cannot take effect

### ISSUE THREAD

**Issue by OneSeven** [wailsapp/wails#916]
**Description**
runtime.WindowMaximise() cannot take effect

**To Reproduce**
Steps to reproduce the behaviour:
Setting runtime.WindowMaximise() anywhere does not take effect，
Maximize first, then restore to the default width and height

**Expected behaviour**
Window is maximized normally

**Screenshots**
No

**System Details**
Go 1.17.2
wails v2.0.0-beta.15
windows10 64bit 21H1

**Additional context**
No

- :ballot_box_with_check: This issue is for Windows and I have read the [Windows Developer Guide](https://wails.app/guides/windows/)

**Comment by leaanthony** [comment]
Works, just not in startup. In the default vanilla project I have this and it works as intended:

```go
// Greet returns a greeting for the given name
func (b *App) Greet(name string) string {
	runtime.WindowMaximise(b.ctx)
	return fmt.Sprintf("Hello %s, It's show time!", name)
}
```

**Comment by OneSeven** [comment]
> Works, just not in startup. In the default vanilla project I have this and it works as intended:
> 
> ```go
> // Greet returns a greeting for the given name
> func (b *App) Greet(name string) string {
> 	runtime.WindowMaximise(b.ctx)
> 	return fmt.Sprintf("Hello %s, It's show time!", name)
> }
> ```

wails v2.0.0-beta.20
The same settings as above, call in JS, nothing happens
Can print out normally
```
func (b *App) WindowMaximise() {
	runtime.WindowMaximise(b.ctx)
	fmt.Println("yyyyy")
}
```
```js
window.go.app.App.WindowMaximise()
```

**Comment by leaanthony** [comment]
I've got a change incoming that should sort this out for you.

**Comment by leaanthony** [comment]
Is this still an issue @OneSeven ?

**Comment by OneSeven** [comment]
> Is this still an issue @OneSeven ?

It is currently valid, but there is a small problem,
It will not take effect when it is initialized.
Call WindowMaximise immediately in Startup, it will be maximized first, and then reduced to the setting values of Width and Height. This situation is the same in JS.
I set the delay function (10s) and it can be maximized normally.

**Comment by redraskal** [comment]
Tested on Windows 11- using `WindowStartState:  options.Maximised` in app options is 100% functional.

Found out the calls to Show in [navigationCompleted](https://github.com/wailsapp/wails/blob/0d402492db8dd36d455d745ed5f7a1a980ab7000/v2/internal/frontend/desktop/windows/frontend.go#L480) were negating the maximize request in startup.

Here's a PR to fix this issue so both options work: #1069 

### Alternate solution:
Set the StartHidden option to true in app options & use the following func
```
func (b *App) startup(ctx context.Context) {
	runtime.WindowShow(ctx)
	runtime.WindowMaximise(ctx)
}
```
